src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,155 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import React from 'react';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (value: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [newMovie, setNewMovie] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+
+  const [errors, setErrors] = useState({
+    title: false,
+    imgUrl: false,
+    imdbUrl: false,
+    imdbId: false,
+  });
+
+  const handleRequiredFields = () => {
+    if (
+      newMovie.title &&
+      newMovie.imgUrl &&
+      newMovie.imdbUrl &&
+      newMovie.imdbId
+    ) {
+      return false;
+    }
+
+    return true;
+  };
+
+  const reset = () => {
+    setNewMovie({
+      title: '',
+      description: '',
+      imgUrl: '',
+      imdbUrl: '',
+      imdbId: '',
+    });
+  };
+
+  const handleChange = (
+    fieldName: keyof typeof newMovie,
+    e: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    const { value } = e.target;
```

Comment:
```suggestion
    const { value, name } = e.target;
```
as Denys suggested you can get the name from the event. In case you have a type error you can use typecasting here: `name as keyof typeof newMovie`